### ISSUE TITLE
Can't login to Garmin

### ISSUE THREAD

**Issue by orutra11** [tcgoetz/GarminDB#209]
After creating a clean conda python 3.11 environment and installing with pip3

`garmindb_cli.py --all --download --import --analyze`

returns a Garth authentication error:

```
garmindb_cli.py --all --download --import --analyze
___Downloading All Data___
Traceback (most recent call last):
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/lib/python3.11/site-packages/garth/http.py", line 125, in request
    self.last_resp.raise_for_status()
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://sso.garmin.com/sso/signin?id=gauth-widget&embedWidget=true&gauthHost=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&service=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&source=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&redirectAfterAccountLoginUrl=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&redirectAfterAccountCreationUrl=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/bin/garmindb_cli.py", line 366, in <module>
    main(sys.argv[1:])
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/bin/garmindb_cli.py", line 347, in main
    download_data(args.overwrite, args.latest, args.stats)
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/bin/garmindb_cli.py", line 112, in download_data
    if not download.login():
           ^^^^^^^^^^^^^^^^
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/lib/python3.11/site-packages/garmindb/download.py", line 69, in login
    self.garth.login(username, password)
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/lib/python3.11/site-packages/garth/http.py", line 140, in login
    self.oauth1_token, self.oauth2_token = sso.login(*args, client=self)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/lib/python3.11/site-packages/garth/sso.py", line 79, in login
    client.post(
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/lib/python3.11/site-packages/garth/http.py", line 137, in post
    return self.request("POST", *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/arturo/opt/anaconda3/envs/runEnv311/lib/python3.11/site-packages/garth/http.py", line 127, in request
    raise GarthHTTPError(
garth.exc.GarthHTTPError: Error in request: 401 Client Error: Unauthorized for url: https://sso.garmin.com/sso/signin?id=gauth-widget&embedWidget=true&gauthHost=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&service=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&source=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&redirectAfterAccountLoginUrl=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&redirectAfterAccountCreationUrl=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed
```

**Comment by tcgoetz** [comment]
The whole exception stack is in garth, so you should file a [garth](https://github.com/matin/garth) bug and then link it here.

**Comment by orutra11** [comment]
As I installed a clean version through pip in a MacOS machine, I figured
this should be reproducible and might happen to everyone installing 3.5.0.
Even if it is a Garth bug, it breaks garmindb. Is it because of a Garth
update? Have you seen this error before?

Thanks!

**Comment by tcgoetz** [comment]
I have not seen this issue and can not reproduce it.

One downside of replacing the garmindb login code with garth is that I don't know the code. I think we need the garth maintainer to fix this.

**Comment by orutra11** [comment]
Whats your current known working environment?

Thanks again.
Arturo.

**Comment by tcgoetz** [comment]
What was the resolution?

**Comment by orutra11** [comment]
I can really pinpoint it to anything. I deleted everything (sources,
configs, envs…). Recreated, installed garth from source and then GarminDB
from pip and it worked!

**Comment by vinisalazar** [comment]
Hi @tcgoetz, apologies for reviving this issue, but I faced this same problem today.

I did some digging and I believe that this is caused by the fact that the program creates `~/.GarminDb/GarminConnectConfig.json` with the placeholder values for username and password (`joe@shmoe` or whatever). Trying to authenticate with these credentials leads to this error.

I propose that the user and password in the `GarminConnectConfig.json` are empty by default, so that the program errors out with this line:
https://github.com/tcgoetz/GarminDB/blob/2c0a9921df6927c33862ccd39fb53154eebc8db0/garmindb/download.py#L76

It may also be useful to print out the `config_path` variable so that users know precisely where the config file is. I missed the `~` at first, creating the `.GarminDb` directory in my current working directory. After realising that the config file is in `~/.GarminDb`, I could fix the problem.

Best,
Vini
